src/state/slices/opportunitiesSlice/resolvers/idle.ts

```diff
@@ -0,0 +1,99 @@
+import type { AssetId } from '@shapeshiftoss/caip'
+import { fromAssetId, toAssetId } from '@shapeshiftoss/caip'
+import { DefiProvider, DefiType } from 'features/defi/contexts/DefiManagerProvider/DefiCommon'
+import { getIdleInvestor } from 'features/defi/contexts/IdleProvider/idleInvestorSingleton'
+import { logger } from 'lib/logger'
+import { selectAssetById } from 'state/slices/selectors'
+
+import type {
+  GetOpportunityIdsOutput,
+  GetOpportunityMetadataOutput,
+  OpportunitiesState,
+  OpportunityDefiType,
+  StakingId,
+} from '../types'
+import type { ReduxApi } from './types'
+
+const moduleLogger = logger.child({ namespace: ['opportunities', 'resolvers', 'idle'] })
+
+export const idleStakingOpportunitiesMetadataResolver = async ({
+  opportunityType,
+  reduxApi,
+}: {
+  opportunityType: OpportunityDefiType
+  reduxApi: ReduxApi
+}): Promise<{ data: GetOpportunityMetadataOutput }> => {
+  const idleInvestor = getIdleInvestor()
+  const opportunities = await idleInvestor.findAll()
+
+  const { getState } = reduxApi
+  const state: any = getState() // ReduxState causes circular dependency
+
+  const byId = await opportunities.reduce(
+    async (acc: Promise<OpportunitiesState[DefiType.Staking]['byId']>, opportunity) => {
+      const assetId = toAssetId({
+        assetNamespace: 'erc20',
+        assetReference: opportunity.id,
+        chainId: fromAssetId(opportunity.feeAsset.assetId).chainId,
+      })
+
+      const accResolved = await acc
+
+      const asset = selectAssetById(state, assetId)
+
+      // Asset doesn't exist in portfolio, meaning this asset is bogus, e.g these two
+      // https://etherscan.io/address/0xa0154a44c1c45bd007743fa622fd0da4f6d67d57
+      // https://etherscan.io/address/0x5f45a578491a23ac5aee218e2d405347a0fafa8e
+      if (!asset) return accResolved
+
+      const rewardAssetIds = (await opportunity.getRewardAssetIds().catch(error => {
```

Comment:
At the moment, we are only able to get "Best Yield" reward asset IDs, quoting @samster91:

> that method is only available for best yield

At the moment, we simply call `getGovTokens()` contract method on contracts that support it (i.e Best Yield contracts), but there will be some additional lib work needed to make it work for Tranches

This will do the trick until lib is updated, but we should eventually remove the try/catch and try/catch the whole function